index.html

```diff
@@ -12045,15 +12084,23 @@ <h4 id="PrimaryGlobal">4.3.6. [Global] and [PrimaryGlobal]</h4>
                 with the following prose definition:
               </p>
               <ol class="algorithm">
-                <li>Let <var>values</var> be the list of <a class="dfnref" href="#dfn-values-to-iterate-over">values to iterate over</a>.</li>
-                <li>Let <var>len</var> be the length of <var>values</var>.</li>
-                <li>Initialize <var>k</var> to 0.</li>
-                <li>While <var>k</var> &lt; <var>len</var>:
+                <li>Let <var>O</var> be the <strong>this</strong> value.</li>
+                <li>Let <var>pairs</var> be the list of <a class="dfnref" href="#dfn-value-pairs-to-iterate-over">value pairs to iterate over</a>.</li>
+                <li>Let <var>i</var> be 0.</li>
+                <li>While <var>i</var> is less than the length of <var>pairs</var>:
                   <ol>
-                    <li>Let <var>kValue</var> be the value in <var>values</var> at index <var>k</var>.</li>
+                    <li>Let <var>pair</var> be the entry in <var>pairs</var> at index <var>i</var>.</li>
+                    <li>Let <var>key</var> be <var>pair</var>’s key.</li>
+                    <li>Let <var>value</var> be <var>pair</var>’s value.</li>
                     <li><a href="#es-invoking-callback-functions">Invoke</a> <var>callback</var> with <var>thisArg</var>
                       as the <a class="dfnref" href="#dfn-callback-this-value">callback this value</a> and
-                      <var>k</var> and <var>value</var> as its arguments.</li>
+                      <var>value</var>, <var>key</var> and <var>O</var> as its arguments.</li>
+                    <li>Update <var>pairs</var> to the current list of <a class="dfnref" href="#dfn-value-pairs-to-iterate-over">value pairs to iterate over</a>.
+                      <div class="note"><div class="noteHeader">Note</div>
+                        <p>Depending on whether prose accompanying the interface defined this to be a snapshot at the time
```

Comment:
> And in any case pairs could be removed and re-inserted at later indexes and be iterated over multiple times

Yep.  I think the same can happen with Map/Set, actually.  For example, this script:

    var s = new Set([1, 2, 3]); 
    var e = s.values(); 
    console.log(e.next()); 
    s.delete(1); 
    s.add(1); 
    console.log(...e)

logs:

    Object { value: 1, done: false }
    2 3 1


> I can't see why there'd be good reasons to choose differently for different APIs.

Well, performance and simplicity, possibly.

But yes, we could do what you suggest.  I don't believe it would match the Map/Set behavior, though, per above.